From 548564c87db77b2ef20302ce0a1183c0d50e4fe5 Mon Sep 17 00:00:00 2001
From: Silke Hofstra <silke@slxh.eu>
Date: Thu, 24 Oct 2024 15:15:56 +0200
Subject: [PATCH 15/15] resolved: reduce mDNS timeout

Retry mDNS probes using factors of 250 ms instead of 1000 ms.
This results in the total timeout being reduced to 1750 ms instead of 7000ms.
---
 src/resolve/resolved-dns-transaction.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/resolve/resolved-dns-transaction.c b/src/resolve/resolved-dns-transaction.c
index 105ad351b6..25e04c3d6e 100644
--- a/src/resolve/resolved-dns-transaction.c
+++ b/src/resolve/resolved-dns-transaction.c
@@ -1569,8 +1569,9 @@ static usec_t transaction_get_resend_timeout(DnsTransaction *t) {
                         return MDNS_PROBING_INTERVAL_USEC;
 
                 /* See RFC 6762 Section 5.1 suggests that timeout should be a few seconds. */
+                /* Reduce this to 125 ms (at least 875 ms in total) for responsiveness. */
                 assert(t->n_attempts > 0);
-                return (1 << (t->n_attempts - 1)) * USEC_PER_SEC;
+                return (1 << (t->n_attempts - 1)) * 125 * USEC_PER_MSEC;
 
         case DNS_PROTOCOL_LLMNR:
                 return t->scope->resend_timeout;
-- 
2.47.0

